mysql-test/main/mysql_upgrade.test

```diff
@@ -472,4 +472,15 @@ drop table mysql.global_priv;
 rename table mysql.global_priv_bak to mysql.global_priv;
 --remove_file $MYSQLD_DATADIR/mysql_upgrade_info
 
+--echo #
+--echo # [MDEV-16735] Upgrades only work if 'alter_algorithm' is DEFAULT
+--echo # or COPY. Test that the session value 'DEFAULT' in mysql_upgrade
+--echo # properly overrides the potentially incompatible global value.
+--echo #
+
+SET GLOBAL alter_algorithm='INPLACE';
+SHOW GLOBAL VARIABLES LIKE 'alter_algorithm';
+--exec $MYSQL_UPGRADE --force 2>&1
+SET GLOBAL alter_algorithm='DEFAULT';
```

Comment:
Actually here you want DEFAULT unquoted. In case, for what will be horribly readable reasons, the `alter_algorithm` is changed.